data/src/main/java/org/apache/iceberg/data/GenericAppenderFactory.java

```diff
@@ -91,7 +118,13 @@ public FileAppender<Record> newAppender(OutputFile outputFile, FileFormat fileFo
   @Override
   public FileAppender<Record> newAppender(
       EncryptedOutputFile encryptedOutputFile, FileFormat fileFormat) {
-    MetricsConfig metricsConfig = MetricsConfig.fromProperties(config);
+    MetricsConfig metricsConfig;
+    if (table == null) {
+      metricsConfig = MetricsConfig.fromProperties(config);
+    } else {
+      metricsConfig = MetricsConfig.forTable(table);
+    }
+
```

Comment:
This is a bit questionable for me.
Do we really want to disregard the config properties specifically set by the `set`/`setAll` method if the table has provided?
We should decide on the expected behavior and make sure that we don't get wrong configuration. Maybe by throwing an exception in the set methods when the table is set?